app/models/target.rb

```diff
@@ -30,12 +30,34 @@ class Target < ApplicationRecord
 
   validate :limit_number_of_targets
 
+  reverse_geocoded_by :latitude, :longitude
+
+  after_create :check_matching_targets
+
   private
 
   def limit_number_of_targets
     return if user && user.targets.count < 10
 
-    errors.add(:base, I18n
-      .t('api.error.invalid_request.max_target_limit'))
+    errors.add(:base,
+               I18n.t('api.error.invalid_request.max_target_limit'))
+  end
+
+  def check_matching_targets
+    matching_users = []
+    Target.find_each do |each_target|
+      matching_users.push(each_target.user) if
+      each_target.topic_id == topic_id &&
+      each_target.user_id != user_id &&
+      matching_users.exclude?(each_target.user) &&
+      distance_to(each_target) <= radius + each_target.radius
+    end
```

Comment:
since map returns an array you can use it and declare:
`matching_users = Target.where(user_id: user.id)........`
then you don't need to declare `matching_users = []`